torch/_inductor/graph.py

```diff
@@ -130,7 +131,22 @@ def increment_randomness_offset(self, numel):
     def run(self, *args):
         return super().run(*args)
 
+    def disable_cpp_wrapper(self, cond):
+        self._can_use_cpp_wrapper = False
+        if config.debug:
+            print("Set _can_use_cpp_wrapper to False due to", cond)
+
+    def check_buffer_for_cpp_wrapper(self, buffer: ir.ComputedBuffer):
+        if isinstance(buffer, ir.ExternKernel):
+            self.disable_cpp_wrapper("ExternKernel")
+        if isinstance(buffer, ir.ComputedBuffer):
+            if buffer.data.get_reduction_type():
+                self.disable_cpp_wrapper("Reduction")
```

Comment:
What is the issue with reductions?

Reply:
For reductions, I found that the code string that I got in the codegen stage and the execution stage are different due to `\`. 

- The code string getting from `code.getvalue()` (`code` is an `IndentedBuffer`) in codegen stage:
  ```
  #pragma omp declare reduction(argmax : struct IndexValue_1 :\
                  omp_out.value = omp_in.value < omp_out.value ? omp_out.value : omp_in.value,\
                  omp_out.index = omp_in.value < omp_out.value ? omp_out.index : omp_in.index)\
                  initializer(omp_priv = {0, -std::numeric_limits<float>::infinity()})
  ```

- The code string loaded during the execution (`\` will be escaped):
  ```
  #pragma omp declare reduction(argmax : struct IndexValue_1 :                omp_out.value = omp_in.value < omp_out.value ? omp_out.value : omp_in.value,                omp_out.index = omp_in.value < omp_out.value ? omp_out.index : omp_in.index)                  initializer(omp_priv = {0, -std::numeric_limits<float>::infinity()})
  ```

Thus I'm not able to get the same hash value for these two pieces of code.
I could add a function to make the transformation to escape the backslash in the codegen stage.

Reply:
Ah, perhaps we could just remove the newline and have less pretty output code.  We could also compile the kernels seperately and link them.

Reply:
I've removed the newline as in https://github.com/pytorch/pytorch/pull/88561.